### PR TITLE
Dev json surfer

### PR DESCRIPTION
**Pull request by Qawasmeh-omar**
Json surfer integrated. Following the new naming conventions for packages and group IDs

### CODE REVIEW

File: sparql-generate-json/src/main/java/fr/mines_stetienne/ci/sparql_generate/json/ITER_JSONSurfer.java [hunk 1]
```diff
@@ -1,268 +1,303 @@
...
+/**
+ * Iterator function
+ * <a href="http://w3id.org/sparql-generate/iter/JSONSurfer">iter:JSONSurfer</a>
+ * extracts a list of sub-JSON documents of a JSON document, according to a
+ * forward-only JSONPath expression. See https://github.com/jsurfer/JsonSurfer
+ * for the JSONPath syntax specification.
+ *
+ * <p>
+ * See <a href=
+ * "https://w3id.org/sparql-generate/playground.html#ex=example/generate/Example-JSONSurfer">Live
```
Comment (maximelefrancois86) [inline sparql-generate-json/src/main/java/fr/mines_stetienne/ci/sparql_generate/json/ITER_JSONSurfer.java:78]: Please create an example named this way

File: sparql-generate-json/src/main/java/fr/mines_stetienne/ci/sparql_generate/json/ITER_JSONSurfer.java [hunk 2]
```diff
@@ -1,268 +1,303 @@
...
+ * </p>
+ *
+ * <ul>
+ * <li>Param 1: (json): the URI of the JSON document (a URI), or the JSON object
+ * itself (a String);</li>
+ * <li>Param 2: (jsonPath) the JSONPath query;</li>
+ * <li>Param 3 .. N : (auxJsonPath ... ) other JSONPath queries, which will be
+ * executed over the results of the execution of jsonPath, and provide one
+ * result each.</li>
+ * </ul>
```
Comment (maximelefrancois86) [inline sparql-generate-json/src/main/java/fr/mines_stetienne/ci/sparql_generate/json/ITER_JSONSurfer.java:89]: Is this up to date ? is there no "batch" parameter like in CSV ?

File: sparql-generate-json/src/main/java/fr/mines_stetienne/ci/sparql_generate/json/ITER_JSONSurfer.java [hunk 3]
```diff
@@ -1,268 +1,303 @@
...
+ *
+ * The following variables may be bound:
+ *
+ * <ul>
+ * <li>Output 1: (string) sub-JSON document, encoded in a string literal;</li>
+ * <li>Output 2 .. N-1: result of the execution of the auxiliary JsonPath
+ * queries on Output 1, encoded as a boolean, float, double, integer, string, as
+ * it best fits;</li>
+ * <li>Output N: (integer) the position of the result in the list;</li>
```
Comment (maximelefrancois86) [inline sparql-generate-json/src/main/java/fr/mines_stetienne/ci/sparql_generate/json/ITER_JSONSurfer.java:98]: is this correct ?

File: sparql-generate-json/src/main/java/fr/mines_stetienne/ci/sparql_generate/json/ITER_JSONSurfer.java [hunk 4]
```diff
@@ -1,268 +1,303 @@
...
+ * <li>Output N+1: (boolean) true if this result has a next result in the
+ * list.</li>
+ * </ul>
+ *
+ * Output 2 and 3 can be used to generate RDF lists from the input, but the use
```
Comment (maximelefrancois86) [inline sparql-generate-json/src/main/java/fr/mines_stetienne/ci/sparql_generate/json/ITER_JSONSurfer.java:103]: Is this correct ?

File: sparql-generate-json/src/main/java/fr/mines_stetienne/ci/sparql_generate/json/ITER_JSONSurfer.java (hunk 1 above)
Comment (Qawasmeh-omar) [reply sparql-generate-json/src/main/java/fr/mines_stetienne/ci/sparql_generate/json/ITER_JSONSurfer.java:78]: An example is added, and other comments were taken into account.

**Review by maximelefrancois86 — APPROVED** [review]

**Pull request merged**